cmd/promctl/main.go

```diff
@@ -0,0 +1,159 @@
+// Copyright 2015 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+	"text/template"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/prometheus/prometheus/config"
+	"github.com/prometheus/prometheus/promql"
+	"github.com/prometheus/prometheus/util/cli"
+	"github.com/prometheus/prometheus/version"
+)
+
+// CheckConfigCmd validates configuration files.
+func CheckConfigCmd(t cli.Term, args ...string) int {
+	if len(args) == 0 {
+		t.Infof("usage: promctl check-config <files>")
+		return 2
+	}
+	failed := false
+
+	for _, arg := range args {
+		if err := checkConfig(t, arg); err != nil {
+			t.Errorf(fmt.Sprintf("  FAILED: %s", err))
+			failed = true
+		}
+		t.Infof("")
+	}
+	if failed {
+		return 1
+	}
+	return 0
+}
+
+func checkConfig(t cli.Term, filename string) error {
+	t.Infof(fmt.Sprintf("Checking %s", filename))
+
+	if stat, err := os.Stat(filename); err != nil {
+		return fmt.Errorf("cannot get file info")
+	} else if stat.IsDir() {
+		return fmt.Errorf("is a directory")
+	}
+
+	content, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+
+	var cfg config.Config
+	if err := yaml.Unmarshal(content, &cfg); err != nil {
```

Comment:
As a user I would naively expect all included rule files to be checked as well. Basically, "do I have a fully loadable config set that won't make Prometheus crash upon startup?". I don't even know if there's a good reason to offer an option to not check the referenced rule files...